build_runner_core/lib/src/asset/reader.dart

```diff
@@ -43,8 +42,8 @@ class SingleStepReader implements AssetReader {
   final FutureOr<GlobAssetNode> Function(
       Glob glob, String package, int phaseNum) _getGlobNode;
 
-  /// The assets read during this step in sorted order.
-  final assetsRead = SplayTreeSet<AssetId>();
+  /// The assets read during this step.
+  final assetsRead = Set<AssetId>();
```

Comment:
Use `HashSet`? It does not preserve order and is faster than the default `LinkedHashSet`.